File: ConsoleFun_test_Java/ConsoleFun_test.java

// Copyright 2021 Dwij Bavisi <dwijbavisi@gmail.com>                   //
//                                                                     //
// Licensed under the Taciturn Coder License v0.0.1, the "License";    //
// You may not use this file except in compliance with the License.    //
// You may obtain a copy of the License at:                            //
//         https://taciturncoder.github.io/License/v0-0-1              //
//                                                                     //
// See the License for the permissions and limitations.                //

package ConsoleFun_test_Java;

import ConsoleFun.ConsoleFun;
import ConsoleFun.cfopts;

public class ConsoleFun_test {
    public static void main(String args[]) {
        cfopts opts = new cfopts();

        opts.rows = 20;
        opts.cols = 20;

        opts.fg = 0;
        opts.bg = ConsoleFun.COLOR_YELLOW;

        ConsoleFun.emptyrect(1, 1, opts);

        opts.rows -= 4;
        opts.cols -= 4;
        opts.fg = ConsoleFun.COLOR_BLACK;
        opts.bg = ConsoleFun.COLOR_RED;
        ConsoleFun.filledrect(3, 3, opts);

        ConsoleFun.gotorc(10, 15);
        ConsoleFun.setcolor(ConsoleFun.COLOR_CYAN, ConsoleFun.COLOR_BLACK);
        System.out.print("Hello World");
    }
}

File: ConsoleFun_test_C/ConsoleFun_test.c

// Copyright 2021 Dwij Bavisi <dwijbavisi@gmail.com>                   //
//                                                                     //
// Licensed under the Taciturn Coder License v0.0.1, the "License";    //
// You may not use this file except in compliance with the License.    //
// You may obtain a copy of the License at:                            //
//         https://taciturncoder.github.io/License/v0-0-1              //
//                                                                     //
// See the License for the permissions and limitations.                //

#include <stdio.h>
#include "../ConsoleFun.h"

int main()
{
    cfopts opts = ConsoleFun_getopts();

    opts.bg = ConsoleFun_COLOR_YELLOW;
    ConsoleFun_emptyrect(0, 0, opts);

    opts.rows -= 4;
    opts.cols -= 4;
    opts.fg = ConsoleFun_COLOR_BLACK;
    opts.bg = ConsoleFun_COLOR_RED;
    ConsoleFun_filledrect(2, 2, opts);

    ConsoleFun_gotorc(10, 15);
    ConsoleFun_setcolor(ConsoleFun_COLOR_CYAN, ConsoleFun_COLOR_BLACK);
    printf("Hello World!\n\tDIMENSIONS: %dx%d", opts.rows, opts.cols);

    return 0;
}